drivers/flash/Kconfig.mcux

```diff
@@ -50,6 +50,22 @@ config FLASH_MCUX_FLEXSPI_NOR_WRITE_BUFFER
 	  This prevents faults when the data to write would be located on the
 	  flash itself.
 
+config FLASH_MCUX_FLEXSPI_HYPERFLASH
+	bool "MCUX FlexSPI HYPERFLASH driver"
```

Comment:
could you please comment before resolving ?
Just for the reviewer to know the solution to the question.

Reply:
Sure, I hadn't though about that.
I rewrote the Kconfig file to use nor for qspi and hyperflash as default.

Reply:
I can still select FLASH_MCUX_FLEXSPI_HYPERFLASH when building for mimxrt1050_evk_qspi

Reply:
Yes. I wanted the choice to be there if you actually wanted to change the configuration in Kconfig. I set the default to FLASH_MCUX_FLEXSPI_NOR for the mimxrt1050_evk_qspi.

Would it be better if the these options were disabled for boards, like a full black-list?

On main right now, I can select FLASH_MCUX_FLEXSPI_NOR and MX25UM51345G symbols for mimxrt1050_evk. The only dependencies are HAS_MCUX_FLEXSPI and FLASH.

Reply:
as i'm unfamiliar in general with the mimxrt1050 evk board in the non-qspi and qspi variants in general, I can't decide if hypeflash should be blocked on the qspi variant.

But if the commit I linked to earlier:
> To enable the QSPI you have to physically modify the board and this
disables the hyperflash.

is true, and enabling `qspi` physically will alter the board so that hypeflash can no longer be used on that board, then I believe we should not allow the hyperflash with the qspi variant.

I assume that `FLASH_MCUX_FLEXSPI_NOR` and `MX25UM51345G` are both valid choices for non-qspi and qspi board variants, correct ?

Reply:
MX25UM51345G is only for mimxrt685_evk.
NOR was  used in 1024, 1060 and 1064 evks.

It is correct that all the boards have to be modified if you want to change between qspi and hyperflash.

Reply:
so a bigger mess than what I could imagine. :dizzy_face: